lib/std/compress/zstandard/RingBuffer.zig

```diff
@@ -0,0 +1,122 @@
+//! This ring buffer stores read and write indices while being able to utilise the full
+//! backing slice by incrementing the indices modulo twice the slice's length and reducing
+//! indices modulo the slice's length on slice access. This means that whether the ring buffer
+//! if full or empty can be distinguised by looking at the different between the read and write
+//! indices without adding an extra boolean flag or having to reserve a slot in the buffer.
+
+const Allocator = @import("std").mem.Allocator;
+const assert = @import("std").debug.assert;
+
+const RingBuffer = @This();
```

Comment:
This data structure appears to be general-purpose, not specific to zstandard. Instead of `std.compress.zstandard.RingBuffer`, I suggest `std.RingBuffer`.

Reply:
It would be general purpose, however there are a number of ways to make a ring buffer, and the current implementation is missing functionality that would probably be desired for more general use. If you like I can move it to `std.RingBuffer`, make some quick additions I think are useful for general purpose use and file an issue for other improvements to it.

Reply:
The ring buffer is now at `std.RingBuffer` and I mentioned in the top-level doc comments that it is _not_ designed for concurrent reading and writing. It shouldn't be much work (i.e. just sprinkling a few atomic operations in the right places) to make it safe for concurrent usage as a SPSC ring buffer. but I'd have to refresh my knowledge of atomics.

These some changes that might be desired (I'd punt them to a follow-up issue, unless you'd rather have them in now):
  - SPSC thread safety
  - Generic data type rather than just `u8`
  - Reader/Writer interface